### DISCUSSION TITLE
Can you use records for request beans?

### DISCUSSION THREAD

**Discussion by lbkulinski** [micronaut-projects/micronaut-core#8294] [Q&A] [answered]
Hello

Is there anyway to use records for request beans? Let's say I have a `POST` endpoint like this:
```
@Post("/{id}")
@Produces(MediaType.APPLICATION_JSON)
public HttpResponse<?> example(@Valid @RequestBean Bean bean) {
    return HttpResponse.ok(
        Map.of(
            "bean", bean.toString()
        )
    );
} //example
```
It would be nice to be able to declare `Bean` as such:
```
@Introspected
public record Bean(HttpRequest<?> httpRequest, @PathVariable @Nullable @Positive Integer id) {
}
```
Unfortunately, using the above leads to a compilation error:
```
java: The route declares a uri variable named [id], but no corresponding method argument is present
```
To get this to work, it seems I would need to still use a POJO declaration:
```
@Introspected
public final class Bean {
    private final HttpRequest<?> httpRequest;

    @PathVariable
    @Nullable
    @Positive
    private final Integer id;

    public Bean(HttpRequest<?> httpRequest, Integer id) {
        this.httpRequest = httpRequest;

        this.id = id;
    }

    public HttpRequest<?> getHttpRequest() {
        return httpRequest;
    }
    
    public Integer getId() {
        return id;
    }
    
    //... hashCode, equals, toString
}
```
Is there any way to get around this so that I could still use records?

Thanks!

**Comment by graemerocher** [comment, marked as answer]
looks like a bug, please report an issue

**Reply by lbkulinski** [reply]
Thank you! It has been created here: https://github.com/micronaut-projects/micronaut-core/issues/8295

**Reply by lbkulinski** [reply]
Any help with this issue: https://github.com/micronaut-projects/micronaut-core/issues/8295? It would be really nice to use records with `@RequestBean`
